pkg/antctl/client.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package antctl
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+var isAgent = strings.HasPrefix(os.Getenv("POD_NAME"), "antrea-agent")
```

Comment:
does that mean we can only run `antctl` for the agent from the Pod itself? I thought we would be able to do it from the Node using `/var/run/antrea/antctl.sock`.

Reply:
Yes, to cooperate with agent, antctl should be run inside the agent pod. If we run antctl on node, it is hard to tell whether the endpoint of the controller or the node agent the antctl should use.